### PR TITLE
fix(api): fix field-level schema mismatches in Responses models

### PR DESCRIPTION
**Pull request by EleanorWho**
# What does this PR do?
Fixes field-level schema mismatches in the Responses API Pydantic models so the generated OpenAPI spec matches the 
  OpenResponses reference spec (POST /responses). This is PR 1 of the plan in #5780.                                 
                                                                                                                     
  Changes:                                                                                                           
  - Add `ResponseTruncation` enum to replace bare `str` for truncation fields
  - Add deprecated `generate_summary` field to `OpenAIResponseReasoning`
  - Make `parallel_tool_calls non-nullable` bool on the response model
  - Fix `store/stream` to emit correct nullable schema without default
  - Fix `stream_options.include_obfuscation` to be nullable without default
  - Add schema_utils helpers: `remove_default_from_schema`, `flatten_nullable_remove_default`, `move_default_into_anyof`

Refs: #5780 

## Test Plan
- Unit tests pass: `uv run pytest tests/unit/providers/responses/builtin/test_openai_responses_params.py -x`
  - mypy passes                                                                                                      
  - All pre-commit hooks pass (including OpenAI API Coverage, API spec breaking change check)                        
  - Conformance score for Responses improved from 88.9% to 90.7% (25 → 21 schema issues)                             
  - Integration tests (replay mode): `uv run --no-sync ./scripts/integration-tests.sh --stack-config server:ci-tests --setup gpt --suite responses`

### CODE REVIEW

File: src/ogx_api/responses/models.py [hunk 1]
```diff
@@ -62,11 +56,10 @@ class ResponseGuardrailSpec(BaseModel):
 class ResponseStreamOptions(BaseModel):
     """Options that control streamed response behavior."""
 
-    model_config = ConfigDict(extra="forbid")
-
-    include_obfuscation: bool = Field(
-        default=True,
+    include_obfuscation: bool | None = Field(
+        default=None,
```
Comment (skamenan7) [inline src/ogx_api/responses/models.py:60]: I noticed `include_obfuscation` changed from `default=True` to `default=None`, and the streaming code now filters it out when None. This means requests that previously sent `include_obfuscation=True` to the downstream provider now don't send it at all. Is that intentional? If the downstream provider's default differs from `True`, this could change behavior for existing users.

File: src/ogx_api/schema_utils.py [hunk 2]
```diff
@@ -248,6 +248,34 @@ def remove_null_from_anyof(schema: dict[str, Any], *, add_nullable: bool = False
...
+    schema.pop("default", None)
+
+
+def flatten_nullable_remove_default(schema: dict[str, Any]) -> None:
+    """Remove null from anyOf and strip default values from schema."""
+    remove_null_from_anyof(schema)
+    schema.pop("default", None)
+
+
+def move_default_into_anyof(schema: dict[str, Any]) -> None:
```
Comment (skamenan7) [inline src/ogx_api/schema_utils.py]: nit: `move_default_into_anyof` is defined here but doesn't appear to be used anywhere. Is it meant for a follow-up PR in the #5780 series, or can it be dropped to keep the diff smaller?

File: src/ogx_api/responses/models.py [hunk 3]
```diff
@@ -62,11 +56,10 @@ class ResponseGuardrailSpec(BaseModel):
 class ResponseStreamOptions(BaseModel):
     """Options that control streamed response behavior."""
 
-    model_config = ConfigDict(extra="forbid")
-
```
Comment (skamenan7) [inline src/ogx_api/responses/models.py:54]: Looks like `model_config = ConfigDict(extra="forbid")` was removed from `ResponseStreamOptions`. This means unknown fields in `stream_options` won't be rejected anymore. Was this intentional for OpenAI API compatibility, or an accidental side effect? If `extra="forbid"` is still needed, it can coexist with the other changes.

File: src/ogx_api/schema_utils.py [hunk 4]
```diff
@@ -248,6 +248,34 @@ def remove_null_from_anyof(schema: dict[str, Any], *, add_nullable: bool = False
             schema["nullable"] = True
 
 
+def remove_default_from_schema(schema: dict[str, Any]) -> None:
```
Comment (skamenan7) [inline src/ogx_api/schema_utils.py:251]: Can you add unit tests for `remove_default_from_schema` and `flatten_nullable_remove_default`? Edge cases like missing "default" key, empty anyOf, or nested structures would be worth covering. There are existing schema tests in tests/unit/openai_api/ that could host them.

**Review by skamenan7 — CHANGES_REQUESTED** [review]
Nice work on the schema conformance improvement, @EleanorWho . The json_schema_extra callback approach is clean and composable.

File: src/ogx_api/responses/models.py [hunk 5]
```diff
@@ -40,13 +41,6 @@ class ResponseItemInclude(StrEnum):
     reasoning_encrypted_content = "reasoning.encrypted_content"
 
 
-class ResponseTruncation(StrEnum):
```
Comment (leseb) [inline src/ogx_api/responses/models.py:43]: why did this move to src/ogx_api/openai_responses.py?

File: src/ogx_api/responses/models.py [hunk 6]
```diff
@@ -62,11 +56,10 @@ class ResponseGuardrailSpec(BaseModel):
 class ResponseStreamOptions(BaseModel):
     """Options that control streamed response behavior."""
 
-    model_config = ConfigDict(extra="forbid")
```
Comment (leseb) [inline src/ogx_api/responses/models.py:53]: why did this go away?

File: src/ogx_api/schema_utils.py (hunk 2 above)
Comment (EleanorWho) [reply src/ogx_api/schema_utils.py]: You're right. This function was added for later use, I should create it in the next PR.

File: src/ogx_api/responses/models.py (hunk 6 above)
Comment (EleanorWho) [reply src/ogx_api/responses/models.py:53]: In the [OpenResponse spec](https://github.com/ogx-ai/ogx/blob/67e038c055dbe916a2ecca0a82a8ff74f893cb46/docs/static/openresponses-spec.json#L1065), `StreamOptionsParam` does not set `additionalProperties: false`. Pydantic's `extra="forbid"` generates `additionalProperties: false` in the OpenAPI output, which creates a schema mismatch. So removing it will align with the spec and also improve forward compatibility if new fields are added.

File: src/ogx_api/responses/models.py (hunk 3 above)
Comment (EleanorWho) [reply src/ogx_api/responses/models.py:54]: Yes, it's intentional for OpenAI API compatibility.

File: src/ogx_api/responses/models.py (hunk 5 above)
Comment (EleanorWho) [reply src/ogx_api/responses/models.py:43]: To match the [OpenResponse spec](https://github.com/ogx-ai/ogx/blob/67e038c055dbe916a2ecca0a82a8ff74f893cb46/docs/static/openresponses-spec.json#L1148) which defines `truncation` with `[enum](enum: [auto, disabled])`, we need to change `truncation: str | None = None` (openai_reponses.py L816) into `truncation: ResponseTruncation | None = None`, where `ResponseTruncation` is a StrEnum. But if  `ResponseTruncation` stays in models.py, then openai_responses.py would need to import it from models.py, but models.py already imports from openai_responses.py, causing a circular import. So I moved this class to openai_responses.py and import it back in models.py to keep the dependency one-directional.

File: src/ogx_api/responses/models.py (hunk 1 above)
Comment (EleanorWho) [reply src/ogx_api/responses/models.py:60]: I searched `include_obfuscation` in the whole code base, and it's only defined on the model, and no code in the server reads or acts on its value. When passed downstream through `stream_options`, it goes to the Chat Completion API which only recognizes `include_usage` and ignores unknown keys. So changing the default from `True` to `None` has no behavioral impact.

**Review by cdoern — APPROVED** [review]
had the same questions as the other reviewers but seems all of those concerns have been answered/addressed. lgtm!
